lib/rack/server.rb

```diff
@@ -438,7 +438,8 @@ def write_pid
       def check_pid!
         case pidfile_process_status
         when :running, :not_owned
-          $stderr.puts "A server is already running. Check #{options[:pid]}."
+          pid = ::File.read(options[:pid]).to_i
```

Comment:
You technically don't need to call `to_i` here.

Reply:
Thanks. I will fix.